embedding/cpp/flutter_engine.cc

```diff
@@ -0,0 +1,119 @@
+// Copyright 2022 Samsung Electronics Co., Ltd. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "include/flutter_engine.h"
+
+#include "utils.h"
+
+std::unique_ptr<FlutterEngine> FlutterEngine::Create(
+    const std::string& assets_path, const std::string& icu_data_path,
+    const std::string& aot_library_path,
+    const std::optional<std::string>& dart_entrypoint,
+    const std::optional<std::vector<std::string>>& dart_entrypoint_args) {
+  return std::unique_ptr<FlutterEngine>(
+      new FlutterEngine(assets_path, icu_data_path, aot_library_path,
+                        dart_entrypoint, dart_entrypoint_args));
+}
+
+FlutterEngine::FlutterEngine(
+    const std::string& assets_path, const std::string& icu_data_path,
+    const std::string& aot_library_path,
+    const std::optional<std::string>& dart_entrypoint,
+    const std::optional<std::vector<std::string>>& dart_entrypoint_args) {
+  FlutterDesktopEngineProperties engine_prop = {};
+  engine_prop.assets_path = assets_path.c_str();
+  engine_prop.icu_data_path = icu_data_path.c_str();
+  engine_prop.aot_library_path = aot_library_path.c_str();
+
+  // Read engine arguments passed from the tool.
+  Utils::ParseEngineArgs(&engine_arguments_);
+  std::vector<const char*> switches;
+  for (auto& arg : engine_arguments_) {
+    switches.push_back(arg.c_str());
+  }
+  engine_prop.switches = switches.data();
+  engine_prop.switches_count = switches.size();
+
+  engine_prop.entrypoint =
+      dart_entrypoint.has_value() ? dart_entrypoint.value().c_str() : nullptr;
+
+  std::vector<const char*> entrypoint_args;
+  if (dart_entrypoint_args.has_value()) {
+    for (const std::string& arg : dart_entrypoint_args.value()) {
+      entrypoint_args.push_back(arg.c_str());
+    }
+  }
+  engine_prop.dart_entrypoint_argc = entrypoint_args.size();
+  engine_prop.dart_entrypoint_argv = entrypoint_args.data();
+
+  engine_ = FlutterDesktopEngineCreate(engine_prop);
```

Comment:
Previously, when FlutterDesktopEngineCreate() failed, an error log was output. I think we need an exception handling for this.

Reply:
You're right. I'll add it.